packages/foam-vscode/src/features/preview-navigation.ts

```diff
@@ -77,4 +92,120 @@ export const markdownItWithFoamTags = (
 const getFoamTag = (content: string) =>
   `<span class='foam-tag'>${content}</span>`;
 
+export const markdownItWithRemoveLinkReferences = (
+  md: markdownit,
+  workspace: FoamWorkspace
+) => {
+  return md.disable('link').use(linkRuleProcessingWithoutReferences());
+};
+
+function linkRuleProcessingWithoutReferences() {
```

Comment:
I fear this approach is a bit heavy and disconnects us from future improvements in markdown-it.

Have you considered instead filtering the files to remove the section generated by Foam before it's sent to markdown-it? (not sure whether with the available API this is possible, have you already looked at that option?)

Reply:
I've tried to do that, but it seems manipulation can only be done on tokens. The link reference definition is not passed on as a token and as such manipulation is hard, The only thing I could find is to change the rule set of the inline link rules like I did above.

What still can be an option is that we don't display the alias when link references are present in the file. It then just shows `link|alias` in the preview. Which is actually the same as in GitHub without any manipulation of the Markdown files.

Reply:
mmm... I would need to look a bit more deeply into the API to properly help here.

I came across this: https://www.npmjs.com/package/markdown-it-replace-link - could it help?

Reply:
I would need to look at. The order in which all changes are executed seem to be of importance here. As soon as you save a document, the presentation of the link reference gets priority above the foam link regex. 

Fresh eyes probably help here ;-)

Reply:
For what I can see [vscode just reads the document ](https://github.com/microsoft/vscode/blob/77ff6eb03bcfb11d3bad26346f0b5ed593cb2a99/extensions/markdown-language-features/src/markdownEngine.ts#L132) and puts it in the parser of MarkdownIt. Then the tokenisation and rules start to apply. In both projects I don't see an possibility to intervene in this behaviour. For example, to filter out parts of the document before the tokenisation starts.